### PR TITLE
Add sphinx docs

### PR DESCRIPTION
**Pull request by tzumainn**
Adds sphinx docs for integration with readthdocs.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+#
+# import os
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = u'ESI'
+copyright = u'2021, Tzu-Mainn Chen'
```
Comment (larsks) [inline docs/source/conf.py:25]: Should this actually be (c) Red Hat?

Comment (tzumainn) [reply docs/source/conf.py]: ah, that sounds right - fixed!

File: docs/source/index.rst [hunk 2]
```diff
@@ -0,0 +1,18 @@
+ESI Documentation
+=================
+
+Welcome to the documentation for Elastic Secure Infrastructure (ESI)!
+
+ESI is an OpenStack architecture that allows multiple bare metal node owners to collaborate to form a single bare metal cloud. Owners have exclusive use of their nodes, and can also lease their nodes out to lessees. Take a look at our presentation for a good high-level overview:
```
Comment (larsks) [inline docs/source/index.rst:6]: s/architecture/project/, maybe? I'm not sure what "an architecture" is...

File: docs/source/usage/index.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+**Lessee Actions**
+
+Users can view available offers and contract a node from an offer.
+
+* List Offers: ``openstack lease offer list``
+* Create Contract: ``openstack lease contract create --offer <offer_uuid> --start-time <start_time> --end-time <end_time>``
+* List Contracts: ``openstack lease contract list``
+* Show Contract: ``openstack lease contract show <contract_uuid``
+* Delete Contract: ``openstack lease contract delete <contract_uuid``
```
Comment (larsks) [inline docs/source/usage/index.rst:66]: You're missing a closing `>` on the last two items here.

File: docs/source/usage/index.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+Additional ESI CLI Actions
+--------------------------
```
Comment (larsks) [inline docs/source/usage/index.rst:69]: I would just take this out if you're not going to fill it in right now.

**Review by larsks — COMMENTED** [review]
Looks good; suggested a couple of minor changes.

File: docs/source/conf.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+author = u'Tzu-Mainn Chen'
```
[the remark this answers is not in the document]
Comment (tzumainn) [reply docs/source/conf.py]: Yeah, I like that better - fixed!

File: docs/source/index.rst (hunk 2 above)
Comment (tzumainn) [reply docs/source/index.rst]: I think I was faking fancy. Fixed!

File: docs/source/usage/index.rst (hunk 3 above)
Comment (tzumainn) [reply docs/source/usage/index.rst]: Whoops - fixed!

File: docs/source/usage/index.rst (hunk 4 above)
Comment (tzumainn) [reply docs/source/usage/index.rst:69]: Ah, right - I forgot to fill this in. I'll update in a later PR; in the meantime I've removed the two subsections and simply noted that python-esiclient provides additional actions, if that works for you?

**Comment by tzumainn** [comment]
Thanks for the review! I've updated the PR - let me know if there's anything else!

**Review by larsks — APPROVED** [review]

**Pull request merged**